### PR TITLE
Initial Release Packaging

### PR DESCRIPTION
**Pull request by rs-amp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
 
-A library for managing stylitics widgets within modern web frameworks, configuring them with content items from Amplience, as well as a collection of content types and schemas you can install to make integrating Stylitics with Amplience considerably easier.
+![Stylitics Widgets)](./docs/media/stylitics-main.png)
+
+A library for managing Stylitics Widgets within modern web frameworks, configuring them with content items from Amplience, as well as a collection of Content Types and Schemas you can install to make integrating Stylitics with Amplience considerably easier.
+
+## ⚙️ Features
+- Convert Content Item into Stylitics arguments
+- Automatically create Stylitics Widget
+- Import Schemas, Content Types, samples Content Items in a Dynamic Content hub
```
Comment (rs-amp) [inline README.md]:
```suggestion
- Import Schemas, Content Types, sample Content Items into a Dynamic Content hub
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,29 @@
 # @amplience/dc-integration-stylitics
 
-A library for managing stylitics widgets within modern web frameworks, configuring them with content items from Amplience, as well as a collection of content types and schemas you can install to make integrating Stylitics with Amplience considerably easier.
+![Stylitics Widgets)](./docs/media/stylitics-main.png)
+
+A library for managing Stylitics Widgets within modern web frameworks, configuring them with content items from Amplience, as well as a collection of Content Types and Schemas you can install to make integrating Stylitics with Amplience considerably easier.
+
+## ⚙️ Features
+- Convert Content Item into Stylitics arguments
+- Automatically create Stylitics Widget
```
Comment (rs-amp) [inline README.md]:
```suggestion
- Automatically create Stylitics Widgets
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+- Import Schemas, Content Types, samples Content Items in a Dynamic Content hub
+- Use provided Templates for Content Rendering Service visualisation and cards
+
+## 🏁 Quickstart
+
+Setting up the Stylitics integration library is a simple import or include of the integration.
```
Comment (rs-amp) [inline README.md]:
```suggestion
Using the Stylitics integration library requires a simple import from NPM or linking a prebuilt js file.
```

Seems weird having this before the import methods are described.

File: README.md [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+Once imported, the following two methods can be used:
 
 - `fromContentItem`: Convert from a content item containing Stylitics widget configuration into arguments you can provide to create the widget.
-- `createWidget`: Create a stylitics widget with a given set of arguments, including view type and account. Automatically loads necessary scripts for the view type asynchronously, and keeps them loaded for future requests.
+- `createWidget`: Create a Stylitics widget with a given set of arguments, including view type and account. Automatically loads necessary scripts for the view type asynchronously, and keeps them loaded for future requests.
 
-## Usage
+These can be used regardless of implementation type (NodeJS project, Simple HTML/JS/CSS project or even using the Amplience Content Rendering Service)
+
+The examples below show how your would use in each implementation type. We would recommend using the [Amplience DC Delivery SDK](https://github.com/amplience/dc-delivery-sdk-js) to fetch your content
```
Comment (rs-amp) [inline README.md]: #
```suggestion
The examples below show how you would use in each implementation type. We would recommend using the [Amplience DC Delivery SDK](https://github.com/amplience/dc-delivery-sdk-js) to fetch your content
```

File: docs/automation.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
+# Automation
+
+Included in the automation folder are types and example content you can install in your Amplience Dynamic Content and Content Hub instances to get started with Stylitics.
+
+Functionality of the automation includes:
+ - Content Type Schemas
+ - Content Types
+ - Sample Content for demo accounts and demo sku's from Stylitics
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
 - Sample Content for demo accounts and demo SKUs from Stylitics
```

File: docs/automation.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+ - Default Visualisations
+
+## Prerequisites for automation
+
+Details for what is required to automate into your Amplience account as well as other assumptions and options.
+
+## Mandatory
+In order for this automation to work, you must have the following:
+- An Amplience Dynamic Content Account.
+- A valid API key (`clientId` & `clientSecret`) for this hub with a minumum of a 'Developer' [role](https://amplience.com/developers/docs/concepts/permissions/roles/)
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
- A valid API key (`clientId` & `clientSecret`) for this hub with a minimum of a 'Developer' [role](https://amplience.com/developers/docs/concepts/permissions/roles/)
```

File: docs/automation.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+- Your Amplience Dynamic Content `hubId` to import
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
- Your Amplience Dynamic Content `hubId` to import into
```

File: docs/automation.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+## Other assumptions
+
+These are other assumptions for the automation but can be changes to match your Amplience Dynamic Content account setup.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
These are other assumptions for the automation, but can be changes to match your Amplience Dynamic Content account setup.
```

File: docs/automation.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+### Amplience Content Rendering Service
+
+The automation has default visualisations for the Amplience Content Rendering Service. These may not be applicable to you unless you are using in your implementation and you have this available in your Amplience account.
+
+If you _**do**_ have this features enabled and you wish to use, simply
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
If you _**do**_ have this feature enabled and you wish to use CRS visualisations, simply
```

File: docs/automation.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+- A valid API key (`clientId` & `clientSecret`) for this hub with a minumum of a 'Developer' [role](https://amplience.com/developers/docs/concepts/permissions/roles/)
+- Your Amplience Dynamic Content `hubId` to import
+
+## Other assumptions
+
+These are other assumptions for the automation but can be changes to match your Amplience Dynamic Content account setup.
+
+### Amplience Content Rendering Service
+
+The automation has default visualisations for the Amplience Content Rendering Service. These may not be applicable to you unless you are using in your implementation and you have this available in your Amplience account.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
The automation has default visualisations for the Amplience Content Rendering Service (CRS). These may not be applicable to you unless you are using in your implementation and you have this available in your Amplience account.
```

CRS is used throughout the repo, so it's good to define what it stands for.

File: docs/automation.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+If you _**do**_ have this features enabled and you wish to use, simply
+upload the files found in `examples/crs/` folder and publish.
+
+If you _**do not**_ have this feature enabled and will not use your implementation follow [configuring visualisations](#icons-cards--visualisations)
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
If you _**do not**_ have this feature enabled or simply aren't using it in your implementation, follow [configuring visualisations](#icons-cards--visualisations) to remove the CRS entry.
```

File: docs/automation.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+### Default Repository
+The automation assumes that you have a repository in your Dynamic Content hub with a 'name' of `content`. This is default in most account setups.
+
+If you have a different setup, or would like to apply this to multiple repositories simply do the following before running any automation:
+
+1. Browse to `amplience-automation/automation-files/type`
+2. Open up all of the `*.hbs` files
+3. Find the `repositories` object and ammend the name to the repository name (s) in your account
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
3. Find the `repositories` object and amend the name to the repository name (s) in your account
```

File: docs/automation.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+4. Save the files and run the automation
+
+## Running the automation
+The files in `amplience-automation/automation-files` are content types and content type schemas that allow you to configure a Stylitics widget using Amplience content items. Also included are example content items that demonstrate how they can be used.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
The files in `amplience-automation/automation-files` are content types and content type schemas that allow you to configure a Stylitics widget using Amplience content items. The folder also includes example content items that demonstrate how they can be used.
```

File: docs/automation.md [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+The easiest way to import these is through a script included with the project. Use `npm run import` to obtain a list of arguments, and provide them like `npm run import -- --clientId a --clientSecret b --hubId c` to import to a specific hub.
+
+## Automation options
+
+These additional options are not mandatory but may be useful:
+
+|Option|Alias|Meaning|
+|:----|:----|:----|
+|`--schemaBaseUri`| |determines what base URI is used for the schema IDs being imported. If not provided, this will default to `https://demostore.amplience.com/`.|
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
|`--schemaBaseUri`| |determines what base URI is used for the schema IDs being imported. If not provided, this will default to `https://demostore.amplience.com`.|
```

Just need to be accurate here as the URI the user provides can't end with a forward slash.

File: docs/automation.md [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+|`--omitExtensionEcommToolkit`|`-o`|Will automate your schemas without the eComm Toolkit extension so that you can manually type product references instead of browsing and selecting|
+
+## Automation result
+Once the automation has been ran, your account will be populated with content schemas, content types, dependant extensions, cards, icons and visualisations all pointing to defauly Stylitics accounts
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
Once you've run the automation, your account will be populated with content schemas, content types, dependant extensions, cards, icons and visualisations all pointing to default Stylitics accounts.
```

File: docs/automation.md [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+
+## Customisations & Implementation
+
+### Icons, Cards & Visualisations
+
+The automation comes with default icons, cards and visualisations. These are stand-alone examples to enable you to see get started quickly.
+
+When using for your implementation it is likely that you will want to ammend these to point to your application. You can ammend directly in each content-type in the Amplience Dynamic Content UI, or ammend in the automation.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
When using for your implementation it is likely that you will want to amend these to point to your application. You can amend directly in each content-type in the Amplience Dynamic Content UI after importing, or amend in the automation before importing.
```

File: docs/automation.md [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+1. Browse to `amplience-automation/automation-files/type`
+2. Open up all of the `*.hbs` files
+3. Depending on what you would like to change, see the attributes [`icons`](#customising-icons), [`cards`](#customising-cards) or [`visualisations`](#customising-visualisations) object and ammend as appropriate.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
3. Depending on what you would like to change, see the attributes [`icons`](#customising-icons), [`cards`](#customising-cards) or [`visualisations`](#customising-visualisations) object and amend as appropriate.
```

File: docs/automation.md [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+
+![Stylitics Content Types)](../docs/media/stylitics-content-types.png)
+
+The files in `Assets` are all icons used by the content types. You can change to be any https url to an image of your choice. Feel free to use these as a basis.
+
+[Icons documentation](https://amplience.com/developers/docs/dev-tools/guides-tutorials/content-types/#choosing-an-icon)
+
+#### Customising Cards
+
+These are iframed HTML sources loaded into provide a preview of the content when selecting from the library view.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
These are iframed HTML pages which provide a preview of the content when selecting from the library view.
```

File: docs/automation.md [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+
+[Cards documentation](https://amplience.com/developers/docs/dev-tools/guides-tutorials/content-types/#configuring-a-card)
+
+![Stylitics cards)](../docs/media/stylitics-cards.png)
+
+
+#### Customising Visualisations
+
+These are iframed HTML sources loaded when editing content to see the result of your content edits.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
These are iframed HTML pages displayed alongside the Amplience content editor to see the result of your content edits as you make them.
```

File: docs/automation.md [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+
+[Visualisations documentation](https://amplience.com/developers/docs/dev-tools/guides-tutorials/content-types/#setting-up-visualizations)
+
+![Visualisations)](../docs/media/stylitics-crs-viz.png)
+
+### Account
+
+This automation comes with a dropdown and selector for some demo accounts from Stylitics to get you up and running fast and test a technical implementation and rendering in your application.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
This automation comes with a dropdown and selector for some demo accounts from Stylitics for quick prototyping that lets you test a technical implementation and rendering in your application.
```

File: docs/automation.md [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+
+If you are a Stylitics customer, will have different account(s) that you are using.
```
Comment (rs-amp) [inline docs/automation.md]: This seems odd as a standalone paragraph.

File: docs/automation.md [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+        "demo-kidgirls",
+        "demo-labels-classic",
+        "demo-mens",
+        "demo-womens"
+    ],
+    "default": "demo"
+}
+```
+
+Simply ammend to the accounts in use for your Stylitics implementation. You can remove the default value or choose one that applies to you and save.
```
Comment (rs-amp) [inline docs/automation.md]:
```suggestion
Simply replace these with the accounts in use for your Stylitics implementation. You can remove the default value or choose one that applies to you and save.
```

File: docs/faq.md [hunk 23]
```diff
@@ -16,3 +16,15 @@ Exact URLs used are listed in the table below:
 
 > Note: The Generic component loads the right widget as per its selected view type
 
+### I am seeing some funny characters in the HTML rendering of my widget
+
+The Stylitics widgets require the HTML page to have UTF-8 encoding enabled. Simply add the following to the `<head />` of your HTML.
```
Comment (rs-amp) [inline docs/faq.md]:
```suggestion
The Stylitics widgets require the HTML page to have UTF-8 encoding enabled. Simply add the following to the `<head>` tag of your HTML.
```

Self-closing the tag is confusing since you won't be able to put a <meta> tag in it like that.

File: docs/developing+building+locally.md [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+## Dependency versions
+
+This demo appliction was developed and tested with:
+
+- Node version 18.x
+- NPM version 9.x
+
+## Building
+
+`npm run build`
```
Comment (rs-amp) [inline docs/developing+building+locally.md:17]: Probably want `npm i` above this.

File: docs/developing+building+locally.md [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+
+This builds the project in a variety of configurations:
+- `build/main`: es6/commonjs target.
+- `build/module`: esnext target.
+- `dist`: Webpack distributions meant for loading in the browser. These load the module into the global scope as `ampStylitics`.
+
+## Changing code
+
+All files for the integration are in the `/src` folder.
+Once completed with your changes, you will need to run `npm run build` for these changes to be compiled
```
Comment (rs-amp) [inline docs/developing+building+locally.md]:
```suggestion
Once your changes are completed, you will need to run `npm run build` for these changes to be compiled.
```

File: docs/developing+building+locally.md [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+
+Compiled JS files are outputted into the `dist` folder after building the project.
+
+`dist/ampStylitics.browser.umd.min.js` is recommended, but legacy and un-minified options are also provided.
+
+You should host on a HTTPS location that can deal with the demands of requests to this file (ie, CDN).
+
+## Referencing a customised integration in NodeJS projects
+
+You can follow standard principles to either load from a local project or direct from your own personal GIT fork.
```
Comment (rs-amp) [inline docs/developing+building+locally.md]:
```suggestion
You can follow standard principles to either load from a local directory or direct from your own personal Git fork.
```

File: docs/developing+building+locally.md [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+
+### GIT
```
Comment (rs-amp) [inline docs/developing+building+locally.md]:
```suggestion
### Git
```

File: support.md [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+
+Should you want to host files on your own HTTPS hosting environments, please feel free to do so and update any URLs used.
+
+> Note: We also host some example visualisations in an Amplience S3 bucket. In an implementation we would expect these to point to your application.
+
+## Watch
+We recommend that you watch this repository so that you can be notified of updates and releases.
+
+## Customisations
+We expect customers to ammend their configurations to point to Stylitics specific accounts and products.
```
Comment (rs-amp) [inline support.md]:
```suggestion
We expect customers to amend their configurations to point to Stylitics specific accounts and products.
```

File: support.md [hunk 29]
```diff
@@ -0,0 +1,44 @@
+# Support
+
+This integration is built as open source which you are free to use and customise for your implementation. Amplience supports this integration for the expected behaviour and roles between the 3 key parts of the integration:
```
Comment (rs-amp) [inline support.md]: The second sentence is a bit hard to parse.

File: docs/developing+building+locally.md (hunk 26 above)
Comment (rs-amp) [reply docs/developing+building+locally.md]: Git is always stylized as Git.

File: docs/automation.md (hunk 21 above)
Comment (rezakalfane) [reply docs/automation.md]: "If you are a Stylitics customer, the account dropdown will have different account(s) that you are using." maybe?

File: docs/developing+building+locally.md (hunk 24 above)
Comment (rezakalfane) [reply docs/developing+building+locally.md:17]: yes!

File: support.md (hunk 29 above)
Comment (rezakalfane) [reply support.md]: Split into two?

File: docs/automation.md (hunk 21 above)
Comment (rs-amp) [reply docs/automation.md]: "If you are a Stylitics customer, you'll want your own set of accounts to be choices on the content form." ?
